src/Sarif.Driver/Sdk/ExportRulesMetadataCommandBase.cs

```diff
@@ -130,12 +130,16 @@ private void OutputSarifRulesMetada(string outputFilePath, ImmutableArray<IRule>
                 newRule.Id = rule.Id;
                 newRule.Name = rule.Name;
                 newRule.HelpUri = rule.HelpUri;
-                newRule.Properties = rule.Properties;
                 newRule.FullDescription = rule.FullDescription;
                 newRule.MessageFormats = rule.MessageFormats;
 
                 newRule.ShortDescription = rule.ShortDescription;
 
+                foreach (string propertyName in rule.PropertyNames)
+                {
+                    newRule.SetProperty(propertyName, rule.GetProperty(propertyName));
```

Comment:
You'll see below that `SetProperty` creates the underlying dictionary if it doesn't already exist, so you'll see other places where I could get rid of SDK code that created an empty dictionary.